server/apps/publish/archive_publish.py

```diff
@@ -385,6 +377,80 @@ def publish(self, doc, updates, target_media_type=None):
 
         return queued
 
+    def sending_to_digital_subscribers(self, doc):
+        subscribers, subscribers_yet_to_receive = self.get_subscribers(doc)
+        filtered_subscribers = self.filter_subscribers(doc, subscribers, DIGITAL)
+        return len(filtered_subscribers) > 0
+
+    def get_subscribers(self, doc):
+        subscribers, subscribers_yet_to_receive = [], []
+        if self.published_state == 'corrected':
+            query = {'$and': [{'item_id': doc[config.ID_FIELD]}, {'publishing_action': 'published'}]}
+            queued_items = get_resource_service('publish_queue').get(req=None, lookup=query)
+
+            if queued_items.count():
+                # Step 2(b)(i)
+                queued_items = list(queued_items)
+                subscriber_ids = {queued_item['subscriber_id'] for queued_item in queued_items}
+
+                query = {'$and': [{config.ID_FIELD: {'$in': list(subscriber_ids)}}]}
+                subscribers = list(get_resource_service('subscribers').get(req=None, lookup=query))
+                if subscribers:
+                    # Step 2(b)(ii)
+                    active_subscribers = get_resource_service('subscribers').get(req=None, lookup={'is_active': True})
+                    subscribers_yet_to_receive = [a for a in active_subscribers
+                                                  if not any(a[config.ID_FIELD] == s[config.ID_FIELD]
+                                                             for s in subscribers)]
+
+                    if len(subscribers_yet_to_receive) > 0:
+                        # Step 2(b)(iii)
+                        if doc.get('targeted_for'):
+                            subscribers_yet_to_receive = list(self.non_digital(subscribers_yet_to_receive))
+
+        elif self.published_state == 'published':  # Step 2(c)
+            # Step 2(c)(i)
+            subscribers = list(get_resource_service('subscribers').get(req=None, lookup={'is_active': True}))
+
+            # Step 2(c)(ii)
+            if doc.get('targeted_for'):
+                subscribers = list(self.non_digital(subscribers))
+
+        return subscribers, subscribers_yet_to_receive
+
+    def filter_subscribers(self, doc, subscribers, target_media_type):
```

Comment:
I didn't want to change the current publish logic to much so basically I just extracted some functions and reused them.
I will look into optimizing the publishing flow so we don't do unneeded subscriber filtering.